pluto-plugins/plugins/network/core/lib/src/main/java/com/pluto/plugins/network/intercept/NetworkData.kt

```diff
@@ -11,42 +12,86 @@ class NetworkData {
+                body == null ||
+                !body.isLikelyJson
+            ) return null
+            val json = runCatching { JSONObject(body!!.body.toString()) }.getOrNull() ?: return null
+            val query = json.optString("query") ?: return null
+            val variables = json.optJSONObject("variables") ?: JSONObject()
+            val match = graqphlQueryRegex.find(query)?.groupValues ?: return null
+            return GraphqlData(
+                queryType = match[1],
+                queryName = match[2],
+                variables = variables,
+            )
+        }
+
         internal val isGzipped: Boolean
             get() = headers["Content-Encoding"].equals("gzip", ignoreCase = true)
     }
 
     data class Response(
+        val request: Request,
         private val statusCode: Int,
         val body: Body?,
         val headers: Map<String, String?>,
         val sentTimestamp: Long,
         val receiveTimestamp: Long,
         val protocol: String = "",
-        val fromDiskCache: Boolean = false
+        val fromDiskCache: Boolean = false,
     ) {
+        val graphqlErrors: List<String> = parseGraphqlErrors()
+        val hasGraphqlErrors get() = graphqlErrors.isNotEmpty()
+
         internal val status: Status
-            get() = Status(statusCode, mapCode2Message(statusCode))
+            get() = Status(statusCode, getStatusMessage())
         val isSuccessful: Boolean
-            get() = statusCode in 200..299
+            get() = statusCode in 200..299 && !hasGraphqlErrors
         internal val isGzipped: Boolean
             get() = headers["Content-Encoding"].equals("gzip", ignoreCase = true)
+
+        private fun getStatusMessage() = mapCode2Message(statusCode) +
+            if (hasGraphqlErrors) ", Response with errors" else ""
+
+        private fun parseGraphqlErrors(): List<String> {
+            if (request.graphqlData == null ||
+                body == null ||
+                !body.isLikelyJson
+            ) return emptyList()
+            val json = runCatching { JSONObject(body!!.body.toString()) }.getOrNull() ?: return emptyList()
+            val errors = json.optJSONArray("errors") ?: return emptyList()
+            return (0 until errors.length())
+                .mapNotNull { errors.optJSONObject(it)?.optString("message") }
+        }
     }
 
     data class Body(
         val body: CharSequence,
-        val contentType: String
+        val contentType: String,
     ) {
         private val contentTypeInternal: ContentType = ContentType.parse(contentType)
         private val mediaType: String = contentTypeInternal.contentType
         internal val mediaSubtype: String = contentTypeInternal.contentSubtype
         internal val isBinary: Boolean = BINARY_MEDIA_TYPES.contains(mediaType)
         val sizeInBytes: Long = body.length.toLong()
         internal val mediaTypeFull: String = "$mediaType/$mediaSubtype"
+        val isLikelyJson get() = !isBinary && body.startsWith('{')
```

Comment:
cant this be decided based on `mediaType` or `mediaSubtype`?

Reply:
fixed